cdragontoolbox/tftdata.py

```diff
@@ -178,6 +178,14 @@ def parse_sets(self, map22, character_names):
 
     def parse_items(self, map22):
         item_entries = [x for x in map22.entries if x.type == "TftItemData"]
+        trait_entries = [x for x in map22.entries if x.type == "TftTraitData"]
+
+        traits_by_hash = {}
+        for trait in trait_entries:
+            trait_apiName = trait.getv("mName")
+            if "Template" in trait_apiName:
+                continue
```

Comment:
It would be a bit safer to not filter on `Template` at all.

Since we don't list traits directly (contrary to `items`), it is not a problem if we store all of them in `traits_by_hash`.
And it is safer, as `traits_by_hash[h]` might trigger an error if a template trait is listed (unlikely, but we are never sure with Riot).

This would reduce the loop to `traits_by_hash = {trait.path.h: trait.getv("mName") for trait in trait_entries}`